### PR TITLE
[Fix] [Frontend] [Relay] Improve check in flow

### LINKED ISSUE #516: Bug(frontend): daily check in
**Describe the bug**
after daily checkin succeeded, the floating button is still there

**Expected behavior**
After successfully checking in, the float widget should be hidden.
Also on the reputation page, not sure it should display "current reputation" (which means including the reputation changes in the current epoch), or only the "already got reputation" (which means the reputation until latest transitioned epoch). I think it should be synchronized.
(see the screenshot, the above number displayed the current reputation, meanwhile the list of record didn't display the +1 from daily checkin)

**Screenshots**
image

also not sure about this part:
image

### PR DESCRIPTION
**Pull request by Xiawpohr**
## Summary

- Hide notification after user checks in successfully.
- Check user reputation every day.
- Show the reputation history of checking in.

## Linked Issue

close #516

### CODE REVIEW

File: packages/frontend/src/features/reporting/components/DiscardCheckIn/DiscardCheckIn.tsx [hunk 1]
```diff
@@ -5,19 +5,18 @@ import {
...
     open = false,
     onClose = () => {},
+    onConfirm = () => {},
     onCheckIn = () => {},
 }: {
     open?: boolean
     onClose?: () => void
+    onConfirm?: () => void
     onCheckIn?: () => void
 }) {
```
Comment (kittybest) [inline packages/frontend/src/features/reporting/components/DiscardCheckIn/DiscardCheckIn.tsx:19]: 之後refactor的時候 可以把這些function的參數都整理成interface？感覺比較清楚

File: packages/frontend/src/features/reporting/hooks/useNotifyCheckIn/useNotifyCheckIn.ts [hunk 2]
```diff
@@ -1,37 +1,59 @@
 import { useReputationScore } from '@/features/reporting'
 import { useLocalStorage } from '@uidotdev/usehooks'
 import dayjs from 'dayjs'
-import { useEffect, useMemo } from 'react'
+import { useEffect } from 'react'
 
-const DISCARDED_AT = 'discarded-checkin-at'
+const CHECKED_IN_AT = 'checked-in-at'
+const DISCARDED_CHECK_IN_AT = 'discarded-check-in-at'
```
Comment (kittybest) [inline packages/frontend/src/features/reporting/hooks/useNotifyCheckIn/useNotifyCheckIn.ts:7]: 之後refactor的時候把這些固定參數都放在同一個file裡面 比較清楚

**Review by kittybest — COMMENTED** [review]
btw, I can get neither the negative reputation nor the positive reputation for both the reporter and respondant, please check

File: packages/frontend/src/features/reporting/components/DiscardCheckIn/DiscardCheckIn.tsx (hunk 1 above)
Comment (Xiawpohr) [reply packages/frontend/src/features/reporting/components/DiscardCheckIn/DiscardCheckIn.tsx]: fixed~

File: packages/frontend/src/features/reporting/hooks/useNotifyCheckIn/useNotifyCheckIn.ts (hunk 2 above)
Comment (Xiawpohr) [reply packages/frontend/src/features/reporting/hooks/useNotifyCheckIn/useNotifyCheckIn.ts]: fixed~

**Review by kittybest — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**